// lol-js/lib/api/game.js
// Generated by CoffeeScript 1.9.0
(function() {
  var api, assert, ld, matchApi, promiseToCb;

  assert = require('assert');

  ld = require('lodash');

  promiseToCb = require('../utils').promiseToCb;

  matchApi = require('./match');

  api = exports.api = {
    fullname: "game-v1.3",
    name: "game",
    version: "v1.3"
  };

  exports.methods = {
    getRecentGamesForSummonerAsync: function(summonerId, options) {
      var cacheParams, region, requestParams, _ref;
      if (options == null) {
        options = {};
      }
      assert.equal(matchApi.api.version, "v2.2", "match API version has changed.");
      region = (_ref = options.region) != null ? _ref : this.defaultRegion;
      requestParams = {
        caller: "getRecentGamesForSummoner",
        region: region,
        url: (this._makeUrl(region, api)) + "/by-summoner/" + summonerId + "/recent"
      };
      cacheParams = {
        key: api.fullname + "-games-" + region + "-" + summonerId,
        region: region,
        api: api,
        objectType: 'games',
        params: {
          summonerId: summonerId
        }
      };
      return this._riotRequestWithCache(requestParams, cacheParams, {}).then((function(_this) {
        return function(games) {
          if (!options.asMatches) {
            return games;
          } else {
            return _this.Promise.all(games.games.map(function(game) {
              var matchOptions;
              matchOptions = options.asMatches === true ? {
                region: region
              } : ld.extend({}, options.asMatches, {
                region: region
              });
              matchOptions.players = ld.clone(game.fellowPlayers);
              matchOptions.players.push({
                championId: game.championId,
                teamId: game.teamId,
                summonerId: summonerId
              });
              return _this.getMatchAsync(game.gameId, matchOptions);
            })).then(function(matches) {
              games.matches = matches;
              return games;
            });
          }
        };
      })(this));
    }
  };

}).call(this);
